moved/src/state_actor.rs

```diff
@@ -117,27 +132,64 @@ impl<
             gas_fee: base_fee_per_gas,
             l1_fee,
             base_token,
+            block_queries,
+            block_memory,
         }
     }
 
-    pub fn handle_msg(&mut self, msg: StateMessage) {
+    pub fn handle_query(&self, msg: Query) {
+        match msg {
+            Query::ChainId { response_channel } => {
+                response_channel.send(self.genesis_config.chain_id).ok()
+            }
+            Query::GetBalance {
+                address,
+                response_channel,
+                ..
+            } => {
+                // TODO: Support balance from arbitrary blocks
```

Comment:
For a separate PR, but we can get this `TODO` item implemented now with the `block_memory`, right?